core/frontend/src/components/wifi/WifiManager.vue

```diff
@@ -0,0 +1,122 @@
+<template>
+  <v-card
+    elevation="1"
+    width="500"
+  >
+    <v-container>
+      <v-container v-if="current_network">
+        <network-card
+          class="connected-network"
+          @click="openDisconnectionDialog()"
+          :network="current_network"
+        />
+      </v-container>
+      <v-container v-else>
+        <spinning-logo />
+      </v-container>
+
+      <v-container v-if="are_networks_available">
+        <network-card
+          class="available-network"
+          @click="openConnectionDialog(network)"
+          v-for="(network, key) in connectable_networks"
+          :key="key"
+          :network="network"
+        />
+      </v-container>
+      <v-container v-else>
+        <spinning-logo />
+      </v-container>
+    </v-container>
+
+    <connection-dialog
+      v-if="selected_network"
+      v-model="show_connection_dialog"
+      :network="selected_network" 
+    />
+
+    <disconnection-dialog
+      v-if="current_network"
+      v-model="show_disconnection_dialog"
+      :network="current_network"
+      :status="wifi_status"
+    />
+  </v-card>
+</template>
+
+<script lang="ts">
+import Vue from 'vue'
+import { getModule } from 'vuex-module-decorators'
+import WifiStore from '@/store/wifi'
+import ConnectionDialog from './ConnectionDialog.vue'
+import DisconnectionDialog from './DisconnectionDialog.vue'
+import NetworkCard from './NetworkCard.vue'
+import SpinningLogo from '../common/SpinningLogo.vue'
+import { Network, WifiStatus } from '@/types/wifi'
+
+const wifi_store: WifiStore = getModule(WifiStore)
+
+export default Vue.extend({
+  name: 'WifiManager',
+  components: {
+    NetworkCard,
+    SpinningLogo,
+    ConnectionDialog,
+    DisconnectionDialog,
+  },
+  data () {
+    return {
+      selected_network: null as Network | null,
+      show_connection_dialog: false,
+      show_disconnection_dialog: false,
+    }
+  },
+  computed: {
+    are_networks_available (): boolean {
+      return wifi_store.available_networks.length === 0 ? false : true
```

Comment:
As general tip, if you are checking for true/false, you don't need to return true or false, this is a bad practice.

Reply:
True. I usually write the whole phrase as a sanity-check and fix it after, but sometimes I forgot to fix rs